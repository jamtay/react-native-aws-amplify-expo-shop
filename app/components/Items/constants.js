export const ITEM_TYPES = Object.freeze({
  MISSING: 'missing',
  AVAILABLE: 'available',
});
