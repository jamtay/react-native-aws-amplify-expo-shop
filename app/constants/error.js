export const ERROR_MESSAGES = Object.freeze({
  ENTER_NUMBER: 'Invalid queue time, please enter a valid number',
  GENERIC: 'Something went wrong, please try again later',
});
