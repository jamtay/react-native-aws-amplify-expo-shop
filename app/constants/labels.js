export const headerLabels = Object.freeze({
  HEADER: 'Quevid 19',
});

export const searchLabels = Object.freeze({
  NAME: 'Search by name...',
  ADDRESS: 'Search by address...',
});

export const queueTimeLabels = Object.freeze({
  LAST_HOUR: 'Last hour:',
  TODAY: 'Today:',
  LAST_WEEK: 'Last week: ',
  MINUTES: 'minutes',
  N_A: 'n/a',
  LATEST_ENTRY: 'Latest entries:',
});

export const favouriteLabels = Object.freeze({
  FAVOURITES_SECTION_HEADER: 'Here are your favourites',
  NO_FAVOURITES:
    'You currently have no favourites. Please press the heart on a store to add to your favourites',
});

export const explorePageLabels = isStores => {
  return {
    STORE_SEARCH: 'Find a store',
    SEARCH_DESCRIPTION: 'A list of stores matching your search criteria',
    SEARCH_RESULTS: isStores
      ? 'More search results'
      : 'No search results, please enter some search criteria to find results',
  };
};

export const storePageLabels = Object.freeze({
  AVG_Q_TIME: 'Average queue times',
  WEEKS_MISSING: "This week's missing items",
  WEEKS_AVAILABLE: "This week's available items",
  NO_ITEMS: isMissing =>
    `No items have been recorded this week. Record ${
      isMissing ? 'a missing' : 'an available'
    } item for it to appear here`,
});

export const addNewRecordingModal = Object.freeze({
  SUBMIT: 'Submit',
  NEW_QUEUE_TIME: 'How many minutes did you queue for?',
  NEW_MISSING_ITEMS: 'Name an item that is missing',
  NEW_AVAILABLE_ITEMS: 'Name an item that is available',
  SUCCESS_QUEUE_TIME: qTime => `You queued for ${qTime} minutes`,
  SUCCESS_ITEMS: (items, isMissing) =>
    `${isMissing ? 'Missing' : 'Available'} items added including ${items[0]}`,
});
