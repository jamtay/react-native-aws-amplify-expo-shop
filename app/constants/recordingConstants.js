export const RECORDING_TYPES = {
  QUEUE_TIME: 'QUEUE_TIME',
  MISSING_ITEMS: 'MISSING_ITEMS',
  AVAILABLE_ITEMS: 'AVAILABLE_ITEMS',
};

//schema.graphql does have an enum type which works for lists and create. But not for @searchable - https://github.com/aws-amplify/amplify-cli/issues/3248
// enum RecordingType {
//     QUEUE_TIME
//     MISSING_ITEMS
//     AVAILABLE_ITEMS
//   }

export const DEFAULT_SORT_OPTION = Object.freeze({
  sort: {
    field: 'floatTimestamp',
    direction: 'desc',
  },
});

export const MAXIMUM_ITEM_COUNT = 5;
