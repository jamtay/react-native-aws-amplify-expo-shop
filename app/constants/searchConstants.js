export const LIMIT_OPTION = Object.freeze({
  limit: 10,
});

export const DEFAULT_SORT_OPTION = Object.freeze({
  sort: {
    field: 'name',
    direction: 'asc',
  },
});
