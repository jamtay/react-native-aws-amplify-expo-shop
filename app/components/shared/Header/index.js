import React from 'react';
import {Header, Left, Body, Right, Title} from 'native-base';
import {headerLabels} from '../../../constants/labels';

const AppHeader = () => {
  return (
    <Header>
      <Left />
      <Body>
        <Title>{headerLabels.HEADER}</Title>
      </Body>
      <Right />
    </Header>
  );
};

export default AppHeader;
