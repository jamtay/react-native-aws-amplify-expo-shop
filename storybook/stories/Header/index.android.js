import React from 'react';
import {AppHeader} from '../../../app/components/shared/Header';

export default function HeaderStory() {
  return <AppHeader />;
}

HeaderStory.defaultProps = {};

HeaderStory.propTypes = {};
