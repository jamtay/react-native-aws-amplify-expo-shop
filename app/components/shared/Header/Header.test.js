import React from 'react';

import renderer from 'react-test-renderer';
import {render} from 'react-native-testing-library';
import {headerLabels} from '../../../constants/labels';
import Header from './index';

describe('Header component tests', () => {
  describe('snapshot test', () => {
    it('should match stored snapshot', () => {
      const tree = renderer.create(<Header />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('render with properties', () => {
    it('should render with correct header', async () => {
      const {getByText} = render(<Header />);
      expect(await getByText(headerLabels.HEADER)).toBeTruthy();
    });
  });
});
