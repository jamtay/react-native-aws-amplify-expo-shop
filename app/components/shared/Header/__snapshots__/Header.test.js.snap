// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests snapshot test should match stored snapshot 1`] = `
<View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#F8F8F8",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          Quevid 19
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      />
    </View>
  </RCTSafeAreaView>
</View>
`;
